src/lib.rs

```diff
@@ -1,28 +1,35 @@
-pub(crate) mod default;
-pub(crate) mod local;
-pub(crate) mod stdlib;
-pub use self::local::Local;
+#![feature(const_fn)]
+#![cfg_attr(all(not(feature = "std"), not(doctest)), no_std)]
 
-#[cfg(feature = "std")]
-pub(crate) mod thread;
-#[cfg(feature = "std")]
-pub use self::thread::ThreadLocal;
+#[cfg(not(feature = "std"))]
+extern crate alloc;
 
-pub use self::default::DefaultLocality;
+#[doc(hidden)]
+pub mod local;
+mod stdlib;
 
-use stdlib::{any, cell::UnsafeCell, fmt, marker::PhantomData};
+#[cfg(not(feature = "std"))]
+use stdlib::{cell::UnsafeCell, fmt, marker::PhantomData, sync::atomic::Ordering};
 
+#[cfg(not(feature = "std"))]
```

Comment:
hmm, i don't think all the ID stuff should necessarily just be _gone_ when the std feature is set. i'd like to be able to use it in e.g. `sharded_slab`to allow overriding the use of thread IDs for shard identifiers...